### PR TITLE
InterTypes-generated python interface for decapodes.

### PR DESCRIPTION
**Pull request by olynch**
This adds the python files, as well as the original intertype file that was used to generated them.

### CODE REVIEW

File: decapodes/decapodes.it [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+end
+
+@sum Equation begin
+  Eq(lhs::Term, rhs::Term)
+end
+
+struct DecaExpr
+  context::Vector{Judgement}
+  equations::Vector{Equation}
+end
```
Comment (fivegrant) [inline decapodes/decapodes.it:25]:
```suggestion
end

struct Header
    id::Union{String, Nothing}
    description::String,
    name::String,
    model_version::String,
    schema::String
    schema_name::String	
end

struct ASKEMDecapode
  header::Header
  model::DecaExpr
end
```

Comment (fivegrant) [reply decapodes/decapodes.it:25]: Note: Not sure if InterTypes can handle Unions.

Comment (fivegrant) [reply decapodes/decapodes.it:25]: Didn't include annotations in the above. They should probably be there.

**Review by fivegrant — DISMISSED** [review]
Looks good overall!

However, it looks like there's some missing metadata we're expecting as [provided by SyntacticModels](https://algebraicjulia.github.io/SyntacticModels.jl/dev/generated/decapodes_examples/).

I think @p-stokes also added annotations which we'll probably need along with the header.

**Comment by jpfairbanks** [comment]
@olynch, do we need to update this since there has been changes to InterTypes?

File: decapodes/decapodes.it [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    name::String,
+    model_version::String,
+    schema::String
+    schema_name::String	
+end
+
+struct ASKEMDecapode
+  header::Header
+  model::DecaExpr
+end
```
Comment (jpfairbanks) [inline decapodes/decapodes.it:39]: @olynch it looks like we to add the following from the SyntacticModels codebase:

```julia
@sum ASKEMDeca begin
  ASKEMDecaExpr(header::Header, model::Decapodes.DecaExpr, annotations::Vector{Annotation{Symbol,Symbol}})
  ASKEMDecapode(header::Header, model::Decapodes.SummationDecapode, annotations::Vector{Annotation{Symbol,Symbol}})
end
```

**Comment by olynch** [comment]
Updated with latest intertypes/decapodes file.

**Review by jpfairbanks — CHANGES_REQUESTED** [review]
@olynch, it looks like we are still missing the SyntacticModels part. 

https://github.com/AlgebraicJulia/SyntacticModels.jl/blob/0ea6f64fffd6b28824a99d028f5b65d3a0cad4f1/src/decapodes.jl#L12

which depends on https://github.com/AlgebraicJulia/SyntacticModels.jl/blob/0ea6f64fffd6b28824a99d028f5b65d3a0cad4f1/src/amr.jl#L81

**Comment by olynch** [comment]
Freshly generated intertype modules from [DiagrammaticEquations](https://github.com/AlgebraicJulia/DiagrammaticEquations.jl) and [SyntacticModels](https://github.com/AlgebraicJulia/SyntacticModels.jl/).

**Review by jpfairbanks — APPROVED** [review]

**Pull request merged**